### PR TITLE
Initial commit of checkers.js

### PR DESCRIPTION
**Pull request by sandinak**
- can set alternating colors for the squares
- uses a defined width and height that can be manipulated
- can slide them based on the step interval
- can alternate the colors

This has been tested in both the testing tool and QLC+ 4.12.4

### CODE REVIEW

**Comment by mcallegari** [comment]
Hi, I think there's something wrong with the cell calculation, especially if you slide the checker before. Examples

Screenshot_20211204_002937
Screenshot_20211204_003012
Screenshot_20211204_003114

**Comment by sandinak** [comment]
how big was the matrix? ill test tomorrow and get it straight

**Comment by mcallegari** [comment]
I think it is a 64x64 matrix

**Comment by sandinak** [comment]
Found a few issues .. i am patching and testing in my spare time .. will push an update shortly. Thanks.

File: resources/rgbscripts/checkers.js [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+      ["Rose"         , 0x7F3F3F],  //  3
+      ["Coral"        , 0x7F3F1F],  //  4
+      ["Dim Red"      , 0x7F0000],  //  5
+      ["Red"          , 0xFF0000],  //  6
+      ["Orange"       , 0xFF3F00],  //  7
+      ["Dim Orange"   , 0x7F1F00],  //  8
+      ["Goldenrod"    , 0x7F3F00],  //  9
+      ["Gold"         , 0xFF7F00],  // 10
+      ["Yellow"       , 0xFFFF00],  // 11
+      ["Dim Yellow"   , 0x7F7F00],  // 12
```
Comment (mcallegari) [inline resources/rgbscripts/checkers.js:41]: Codacy has a fix for the issue: [Replace `······["Dim·Yellow"···,·0x7F7F00],··` with `····["Dim·Yellow",·0x7f7f00],·`](https://app.codacy.com/gh/mcallegari/qlcplus/pullRequest?prid=9305479)
```suggestion
    ["Dim Yellow", 0x7f7f00], // 12
```

File: resources/rgbscripts/checkers.js [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+      ["Lime"         , 0x7FFF00],  // 13
+      ["Pale Green"   , 0x3F7F00],  // 14
+      ["Dim Green"    , 0x007F00],  // 15
+      ["Green"        , 0x00FF00],  // 16
+      ["Seafoam"      , 0x00FF3F],  // 17
+      ["Turquoise"    , 0x007F3F],  // 18
+      ["Teal"         , 0x007F7F],  // 19
+      ["Cyan"         , 0x00FFFF],  // 20
+      ["Electric Blue", 0x007FFF],  // 21
```
Comment (mcallegari) [inline resources/rgbscripts/checkers.js:50]: Codacy has a fix for the issue: [Replace `······["Electric·Blue",·0x007FFF],·` with `····["Electric·Blue",·0x007fff],`](https://app.codacy.com/gh/mcallegari/qlcplus/pullRequest?prid=9305479)
```suggestion
    ["Electric Blue", 0x007fff], // 21
```

File: resources/rgbscripts/checkers.js [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+var testAlgo;
+
+(
+  function()
+  {
+    var colorPalette = new Object;
+    colorPalette.collection = new Array(
+      ["White"        , 0xFFFFFF],  //  0
+      ["Cream"        , 0xFFFF7F],  //  1
+      ["Pink"         , 0xFF7F7F],  //  2
```
Comment (mcallegari) [inline resources/rgbscripts/checkers.js:31]: Codacy has a fix for the issue: [Replace `······["Pink"·········,·0xFF7F7F],·` with `····["Pink",·0xff7f7f],`](https://app.codacy.com/gh/mcallegari/qlcplus/pullRequest?prid=9305479)
```suggestion
    ["Pink", 0xff7f7f], //  2
```

File: resources/rgbscripts/checkers.js [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+      ["Lime"         , 0x7FFF00],  // 13
+      ["Pale Green"   , 0x3F7F00],  // 14
+      ["Dim Green"    , 0x007F00],  // 15
+      ["Green"        , 0x00FF00],  // 16
+      ["Seafoam"      , 0x00FF3F],  // 17
+      ["Turquoise"    , 0x007F3F],  // 18
+      ["Teal"         , 0x007F7F],  // 19
+      ["Cyan"         , 0x00FFFF],  // 20
+      ["Electric Blue", 0x007FFF],  // 21
+      ["Blue"         , 0x0000FF],  // 22
```
Comment (mcallegari) [inline resources/rgbscripts/checkers.js:51]: Codacy has a fix for the issue: [Replace `······["Blue"·········,·0x0000FF],·` with `····["Blue",·0x0000ff],`](https://app.codacy.com/gh/mcallegari/qlcplus/pullRequest?prid=9305479)
```suggestion
    ["Blue", 0x0000ff], // 22
```

File: resources/rgbscripts/checkers.js [hunk 5]
```diff
@@ -0,0 +1,360 @@
...
+    var colorPalette = new Object;
+    colorPalette.collection = new Array(
+      ["White"        , 0xFFFFFF],  //  0
+      ["Cream"        , 0xFFFF7F],  //  1
+      ["Pink"         , 0xFF7F7F],  //  2
+      ["Rose"         , 0x7F3F3F],  //  3
+      ["Coral"        , 0x7F3F1F],  //  4
+      ["Dim Red"      , 0x7F0000],  //  5
+      ["Red"          , 0xFF0000],  //  6
+      ["Orange"       , 0xFF3F00],  //  7
```
Comment (mcallegari) [inline resources/rgbscripts/checkers.js:36]: Codacy has a fix for the issue: [Replace `······["Orange"·······,·0xFF3F00],·` with `····["Orange",·0xff3f00],`](https://app.codacy.com/gh/mcallegari/qlcplus/pullRequest?prid=9305479)
```suggestion
    ["Orange", 0xff3f00], //  7
```

File: resources/rgbscripts/checkers.js [hunk 6]
```diff
@@ -0,0 +1,360 @@
...
+    colorPalette.makeSubArray = function(_index)
+    {
+      var _array = new Array();
+      for (var i = 0; i < colorPalette.collection.length; i++)
+      {
+        _array.push(colorPalette.collection[i][_index]);
+      }
+      return _array;
+    };
+    colorPalette.names  = colorPalette.makeSubArray(0);
```
Comment (mcallegari) [inline resources/rgbscripts/checkers.js:71]: Codacy has a fix for the issue: [Replace `····colorPalette.names·` with `··colorPalette.names`](https://app.codacy.com/gh/mcallegari/qlcplus/pullRequest?prid=9305479)
```suggestion
  colorPalette.names = colorPalette.makeSubArray(0);
```

File: resources/rgbscripts/checkers.js [hunk 7]
```diff
@@ -0,0 +1,360 @@
...
+      ["Dim Green"    , 0x007F00],  // 15
+      ["Green"        , 0x00FF00],  // 16
+      ["Seafoam"      , 0x00FF3F],  // 17
+      ["Turquoise"    , 0x007F3F],  // 18
+      ["Teal"         , 0x007F7F],  // 19
+      ["Cyan"         , 0x00FFFF],  // 20
+      ["Electric Blue", 0x007FFF],  // 21
+      ["Blue"         , 0x0000FF],  // 22
+      ["Dim Blue"     , 0x00007F],  // 23
+      ["Pale Blue"    , 0x1F1F7F],  // 24
```
Comment (mcallegari) [inline resources/rgbscripts/checkers.js:53]: Codacy has a fix for the issue: [Replace `······["Pale·Blue"····,·0x1F1F7F],·` with `····["Pale·Blue",·0x1f1f7f],`](https://app.codacy.com/gh/mcallegari/qlcplus/pullRequest?prid=9305479)
```suggestion
    ["Pale Blue", 0x1f1f7f], // 24
```

File: resources/rgbscripts/checkers.js [hunk 8]
```diff
@@ -0,0 +1,360 @@
...
+// Development tool access
+var testAlgo;
+
+(
+  function()
+  {
+    var colorPalette = new Object;
+    colorPalette.collection = new Array(
+      ["White"        , 0xFFFFFF],  //  0
+      ["Cream"        , 0xFFFF7F],  //  1
```
Comment (mcallegari) [inline resources/rgbscripts/checkers.js:30]: Codacy has a fix for the issue: [Replace `··["Cream"········,·0xFFFF7F],·` with `["Cream",·0xffff7f],`](https://app.codacy.com/gh/mcallegari/qlcplus/pullRequest?prid=9305479)
```suggestion
    ["Cream", 0xffff7f], //  1
```

File: resources/rgbscripts/checkers.js [hunk 9]
```diff
@@ -0,0 +1,360 @@
...
+      ["Yellow"       , 0xFFFF00],  // 11
+      ["Dim Yellow"   , 0x7F7F00],  // 12
+      ["Lime"         , 0x7FFF00],  // 13
+      ["Pale Green"   , 0x3F7F00],  // 14
+      ["Dim Green"    , 0x007F00],  // 15
+      ["Green"        , 0x00FF00],  // 16
+      ["Seafoam"      , 0x00FF3F],  // 17
+      ["Turquoise"    , 0x007F3F],  // 18
+      ["Teal"         , 0x007F7F],  // 19
+      ["Cyan"         , 0x00FFFF],  // 20
```
Comment (mcallegari) [inline resources/rgbscripts/checkers.js:49]: Codacy has a fix for the issue: [Replace `··["Cyan"·········,·0x00FFFF],··` with `["Cyan",·0x00ffff],·`](https://app.codacy.com/gh/mcallegari/qlcplus/pullRequest?prid=9305479)
```suggestion
    ["Cyan", 0x00ffff], // 20
```

File: resources/rgbscripts/checkers.js [hunk 10]
```diff
@@ -0,0 +1,360 @@
...
+      ["Purple"       , 0x3F00BF],  // 26
+      ["Violet"       , 0x7F007F],  // 27
+      ["Magenta"      , 0xFF00FF],  // 28
+      ["Hot Pink"     , 0xFF003F],  // 29
+      ["Deep Pink"    , 0x7F001F],  // 30
+      ["OFF"          , 0x000000]); // 31
+    
+    colorPalette.makeSubArray = function(_index)
+    {
+      var _array = new Array();
```
Comment (mcallegari) [inline resources/rgbscripts/checkers.js:64]: Codacy has a fix for the issue: [Delete `··`](https://app.codacy.com/gh/mcallegari/qlcplus/pullRequest?prid=9305479)
```suggestion
    var _array = new Array();
```

File: resources/rgbscripts/checkers.js [hunk 11]
```diff
@@ -0,0 +1,360 @@
...
+  {
+    var colorPalette = new Object;
+    colorPalette.collection = new Array(
+      ["White"        , 0xFFFFFF],  //  0
+      ["Cream"        , 0xFFFF7F],  //  1
+      ["Pink"         , 0xFF7F7F],  //  2
+      ["Rose"         , 0x7F3F3F],  //  3
+      ["Coral"        , 0x7F3F1F],  //  4
+      ["Dim Red"      , 0x7F0000],  //  5
+      ["Red"          , 0xFF0000],  //  6
```
Comment (mcallegari) [inline resources/rgbscripts/checkers.js:35]: Codacy has a fix for the issue: [Replace `··["Red"··········,·0xFF0000],··` with `["Red",·0xff0000],·`](https://app.codacy.com/gh/mcallegari/qlcplus/pullRequest?prid=9305479)
```suggestion
    ["Red", 0xff0000], //  6
```

File: resources/rgbscripts/checkers.js [hunk 12]
```diff
@@ -0,0 +1,360 @@
...
+      ["Orange"       , 0xFF3F00],  //  7
+      ["Dim Orange"   , 0x7F1F00],  //  8
+      ["Goldenrod"    , 0x7F3F00],  //  9
+      ["Gold"         , 0xFF7F00],  // 10
+      ["Yellow"       , 0xFFFF00],  // 11
```
Comment (mcallegari) [inline resources/rgbscripts/checkers.js:40]: Codacy has a fix for the issue: [Replace `······["Yellow"·······,·0xFFFF00],·` with `····["Yellow",·0xffff00],`](https://app.codacy.com/gh/mcallegari/qlcplus/pullRequest?prid=9305479)
```suggestion
    ["Yellow", 0xffff00], // 11
```

File: resources/rgbscripts/checkers.js [hunk 13]
```diff
@@ -0,0 +1,360 @@
...
+      ["Dim Yellow"   , 0x7F7F00],  // 12
+      ["Lime"         , 0x7FFF00],  // 13
```
Comment (mcallegari) [inline resources/rgbscripts/checkers.js:42]: Codacy has a fix for the issue: [Replace `······["Lime"·········,·0x7FFF00],·` with `····["Lime",·0x7fff00],`](https://app.codacy.com/gh/mcallegari/qlcplus/pullRequest?prid=9305479)
```suggestion
    ["Lime", 0x7fff00], // 13
```

File: resources/rgbscripts/checkers.js [hunk 14]
```diff
@@ -0,0 +1,360 @@
...
+      ["Green"        , 0x00FF00],  // 16
+      ["Seafoam"      , 0x00FF3F],  // 17
+      ["Turquoise"    , 0x007F3F],  // 18
+      ["Teal"         , 0x007F7F],  // 19
+      ["Cyan"         , 0x00FFFF],  // 20
+      ["Electric Blue", 0x007FFF],  // 21
+      ["Blue"         , 0x0000FF],  // 22
+      ["Dim Blue"     , 0x00007F],  // 23
+      ["Pale Blue"    , 0x1F1F7F],  // 24
+      ["Indigo"       , 0x1F00BF],  // 25
```
Comment (mcallegari) [inline resources/rgbscripts/checkers.js:54]: Codacy has a fix for the issue: [Replace `······["Indigo"·······,·0x1F00BF],·` with `····["Indigo",·0x1f00bf],`](https://app.codacy.com/gh/mcallegari/qlcplus/pullRequest?prid=9305479)
```suggestion
    ["Indigo", 0x1f00bf], // 25
```

File: resources/rgbscripts/checkers.js [hunk 15]
```diff
@@ -0,0 +1,360 @@
...
+      ["Purple"       , 0x3F00BF],  // 26
+      ["Violet"       , 0x7F007F],  // 27
+      ["Magenta"      , 0xFF00FF],  // 28
+      ["Hot Pink"     , 0xFF003F],  // 29
+      ["Deep Pink"    , 0x7F001F],  // 30
+      ["OFF"          , 0x000000]); // 31
+
```
Comment (mcallegari) [inline resources/rgbscripts/checkers.js:61]: Codacy has a fix for the issue: [Delete `····`](https://app.codacy.com/gh/mcallegari/qlcplus/pullRequest?prid=9305479)
```suggestion

```

File: resources/rgbscripts/checkers.js [hunk 16]
```diff
@@ -0,0 +1,360 @@
...
+      ["Cream"        , 0xFFFF7F],  //  1
+      ["Pink"         , 0xFF7F7F],  //  2
+      ["Rose"         , 0x7F3F3F],  //  3
+      ["Coral"        , 0x7F3F1F],  //  4
+      ["Dim Red"      , 0x7F0000],  //  5
+      ["Red"          , 0xFF0000],  //  6
+      ["Orange"       , 0xFF3F00],  //  7
+      ["Dim Orange"   , 0x7F1F00],  //  8
+      ["Goldenrod"    , 0x7F3F00],  //  9
+      ["Gold"         , 0xFF7F00],  // 10
```
Comment (mcallegari) [inline resources/rgbscripts/checkers.js:39]: Codacy has a fix for the issue: [Replace `··["Gold"·········,·0xFF7F00],·` with `["Gold",·0xff7f00],`](https://app.codacy.com/gh/mcallegari/qlcplus/pullRequest?prid=9305479)
```suggestion
    ["Gold", 0xff7f00], // 10
```

**Comment by mcallegari** [comment]
@sandinak any update on this and flames? Thanks

**Comment by sandinak** [comment]
> Isn't this quite similar (if not the same) to what alternate.js can achieve, maybe except for the stepwise sliding? 

Yes and no. We had a show that the basis was "Chess" so I needed a board vs stripes.  This SHOULD be making a checkerboard on the matrix with sizes as indicated, so I'll resolve the issues.

**Comment by hjtappe** [comment]
The chessboard is available in alternate.js in the "Interleaved" Orientation. You may want to check that out.

A note on codacy: It is misleading on the JS code, e.g. regarding "var" versus "let" where practically the warning is valid, but the ECMAscript engine in QT does not digest it and fails with errors.

image

**Comment by mcallegari** [comment]
Sometimes Codacy is inappropriate so you can ignore all those fake errors.
I'm more interested in not introducing duplicates of existing scripts.
If @sandinak got convinced that alternate.js can do the trick, let's please close this.

**Pull request closed without merging**